packages/near-fast-auth-signer-e2e-tests/tests/sign/sign-transaction.spec.ts

```diff
@@ -0,0 +1,128 @@
+const userLAK = KeyPair.fromRandom('ed25519');
+let accountId;
+
+describe('Sign transaction', () => {
+  beforeAll(async ({ browser }, { workerIndex }) => {
+    const context = await browser.newContext();
+    page = await context.newPage();
+    const testDapp = new TestDapp(page);
+    const user = `testuser-playwright-fastauth-${workerIndex}-${Date.now()}`;
+    const email = `${user}@example.com`;
+    accountId = `${user}.testnet`;
+    const oidcToken = await generateNewUserToken(email);
+    const frpKeypair = await claimToken(oidcToken);
+    await createNEARAccount({
+      accountId,
+      fullAccessKeys:    [userFAK.getPublicKey().toString()],
+      limitedAccessKeys: [{
+        public_key:   userLAK.getPublicKey().toString(),
+        receiver_id:  accountId,
+        allowance:    '100000000000000000000000000',
+        method_names: '["claim"]',
+      }],
+      accessToken: oidcToken,
+      oidcKeypair: frpKeypair,
+    });
+    await page.goto('/');
+    await page.waitForLoadState('domcontentloaded');
+    await testDapp.loginWithKeyPairLocalStorage(accountId, userLAK, userFAK);
+    // eslint-disable-next-line no-shadow
+    await page.evaluate(([accountId]) => {
+      window.localStorage.setItem('transactionData', JSON.stringify({
+        receiverId: 'v1.social08.testnet',
+        actions:    [
+          {
+            type:   'FunctionCall',
+            params: {
+              methodName: 'set',
+              args:       {
+                data: {
+                  [accountId]: {
+                    'fast-auth-e2e-test': 'true'
+                  }
+                }
+              },
+              gas:     '300000000000000',
+              deposit: '20000000000000000000000'
+            }
+          }
+        ]
+      }));
+    }, [accountId]);
+  });
+
+  test('should display signerId, transaction amount, receiverId and args', async () => {
+    await page.goto('/');
+    const walletSelector = page.locator('#ws-loaded');
+    await walletSelector.waitFor();
+    await page.locator('data-test-id=sign-transaction-button').click();
+
+    await expect(getFastAuthIframe(page).getByText(accountId)).toBeVisible();
+    await expect(getFastAuthIframe(page).getByText('0.02 NEAR')).toBeVisible();
+
+    await getFastAuthIframe(page).locator('data-test-id=more-details-button').click();
+    await getFastAuthIframe(page).locator('data-test-id=function-call-button').click();
+
+    await expect(getFastAuthIframe(page).getByText(/v1.social08.testnet/)).toBeVisible();
+    await expect(getFastAuthIframe(page).getByText(/"fast-auth-e2e-test": "true"/)).toBeVisible();
+  });
+
+  test('should fail if signer app is not authenticated', async () => {
+    await page.goto('/');
+    await setupPasskeysFunctions(page, 'page', {
+      isPassKeyAvailable:  false,
+      shouldCleanStorage:  false,
+    });
+
+    const walletSelector = page.locator('#ws-loaded');
+    await walletSelector.waitFor();
```

Comment:
To simplify a bit this repeating pattern I think we can do it on a single line:

`await expect(walletSelector).toBeVisible();`

Reply:
But we still have to wait for that element to be there since it's a new page each time right? The difference is syntactical?

Reply:
Yes, but it will also make it explicit that we are waiting for the element to be visible. With `waitFor`, it's unclear what we are waiting for. 

We can also extract it to a function as it's commonly used.